kafka-connect-aws-s3/src/main/scala/io/lenses/streamreactor/connect/aws/s3/sink/S3WriterManager.scala

```diff
@@ -52,14 +53,14 @@ case class MapKey(topicPartition: TopicPartition, partitionValues: immutable.Map
   * sinks, since file handles cannot be safely shared without considerable overhead.
   */
 class S3WriterManager(
-  commitPolicyFn:       TopicPartition => Either[SinkError, CommitPolicy],
-  bucketAndPrefixFn:    TopicPartition => Either[SinkError, S3Location],
-  fileNamingStrategyFn: TopicPartition => Either[SinkError, S3FileNamingStrategy],
-  stagingFilenameFn:    (TopicPartition, Map[PartitionField, String]) => Either[SinkError, File],
-  finalFilenameFn:      (TopicPartition, Map[PartitionField, String], Offset) => Either[SinkError, S3Location],
-  formatWriterFn:       (TopicPartition, File) => Either[SinkError, S3FormatWriter],
-  indexManager:         IndexManager,
-  transformerF:         MessageDetail => Either[RuntimeException, MessageDetail],
+  commitPolicyFn:    TopicPartition => Either[SinkError, CommitPolicy],
```

Comment:
would it make sense to create an interface which can provide the info for : commitPolicyFn, bucketAndPRefixFn, and Key namer in one method .

The same with